src/query/storages/system/src/columns_table.rs

```diff
@@ -194,23 +194,35 @@ pub(crate) async fn dump_tables(
         }
     }
 
+    let visibility_checker = ctx.get_visibility_checker().await?;
+
+    let mut final_dbs: Vec<(String, u64)> = Vec::new();
+
     if databases.is_empty() {
         let all_databases = catalog.list_databases(tenant.as_str()).await?;
         for db in all_databases {
-            databases.push(db.name().to_string());
+            let db_id = db.get_db_info().ident.db_id;
+            let db_name = db.name();
+            if visibility_checker.check_database_visibility(CATALOG_DEFAULT, db_name, db_id)
+                || db_name.to_lowercase() == "information_schema"
```

Comment:
how about moving the `db_name.to_lowercase() == "information_schema"` check into `check_database_visibility`?

IMHO this check is an invariant on calling `check_database_visibility`, it'd be easier to keep this semantic consistent across different calling sites.

Reply:
Done